subprojects/core/src/main/java/org/gradle/internal/operations/trace/BuildOperationTrace.java

```diff
@@ -338,10 +341,15 @@ public Void getResult() {
             assert pendings.isEmpty();
 
             return roots;
+        } catch (AssertionError e) {
+            try {
+                throw new AssertionError(Files.asCharSource(logFile, Charsets.UTF_8).read(), e);
```

Comment:
How large the `logFile` could be? Is it ok to put all its content to exception message?